aroma/utils.py

```diff
@@ -6,6 +6,40 @@
 import nibabel as nib
 import numpy as np
 from nilearn import image, masking
+from scipy import ndimage
+
+
+def derive_masks(in_file, csf=None):
+    if csf is None:
+        print("No CSF TPM/mask provided. Using packaged masks.")
+        mask_dir = get_resource_path()
+        csf_img = nib.load(op.join(mask_dir, "mask_csf.nii.gz"))
+        out_img = nib.load(op.join(mask_dir, "mask_out.nii.gz"))
+        brain_img = image.math_img("1 - mask", mask=out_img)
+        edge_img = nib.load(op.join(mask_dir, "mask_edge.nii.gz"))
+    else:
+        brain_img = masking.compute_epi_mask(in_file)
+        out_img = image.math_img("1 - mask", mask=brain_img)
+        csf_img = nib.load(csf)
+        csf_data = csf_img.get_fdata()
+        if len(np.unique(csf_data)) == 2:
+            print("CSF mask provided. Inferring other masks.")
+        else:
+            print("CSF TPM provided. Inferring CSF and other masks.")
+            csf_img = image.math_img("csf >= 0.95", csf=csf_img)
```

Comment:
Per the ICA-AROMA paper:

> To this end we defined a CSF mask by thresholding a CSF segmentation prior, supplied as part of FSL, at 95% of the robust range.

@CesarCaballeroGaudes does that look good to you?

Reply:
Yes, 95% looks good to me, and even 90%

Reply:
Yes, 95% looks good. Even 90% would also be fine.